src/main/java/com/welovecoding/nbeditorconfig/io/reader/FileInfoReader.java

```diff
@@ -163,9 +164,49 @@ private static String readFirstLineWithSeparator(FileObject fo, Charset charset)
     return firstLine;
   }
 
-  public static String trimTrailingWhitespace(Stream<String> lines, String lineEnding) {
-    return lines.map((String content) -> {
-      return content.replaceAll("\\s+$", "");
-    }).collect(Collectors.joining(lineEnding));
+  public static String trimTrailingWhitespace(Collection<String> lines, String lineEnding) {
+    StringBuilder sb = new StringBuilder();
+    for (String content : lines) {
+      sb.append(content.replaceAll("\\s+$", ""));
+      sb.append(lineEnding);
+    }
+    return sb.toString().trim();
+  }
+
+  public static String trimTrailingWhitespace(String text, String lineEnding) {
+    List<String> lines = readLines(text);
+    return trimTrailingWhitespace(lines, lineEnding);
+  }
+
+  public static String replaceLineEndings(Collection<String> lines, String lineEnding) {
+    StringBuilder sb = new StringBuilder();
+    for (String content : lines) {
+      sb.append(content);
+      sb.append(lineEnding);
+    }
+    return sb.toString().trim();
+  }
+
+  public static String replaceLineEndings(String text, String lineEnding) {
+    List<String> lines = readLines(text);
+    return replaceLineEndings(lines, lineEnding);
+  }
+
+  public static List<String> readLines(String text) {
+    List<String> lines = new ArrayList<>();
+    try (BufferedReader reader = new BufferedReader(new StringReader(text))) {
+
+      try {
+        for (String line = reader.readLine(); line != null; line = reader.readLine()) {
```

Comment:
I think nested expressions in `while` statements are not that readable.
Personally I am more of a for loop through `BufferedReader.lines` kinda guy.
But I won't stop you if you insist on it.